src/utils/requests.js

```diff
@@ -0,0 +1,53 @@
+const base_url = "https://localhost:8000";
+
+const sendRequest = async (url, method, body, headers = {}) => {
+  try {
+    const response = await fetch(base_url + url, {
+      method,
+      body,
+      headers,
+    });
+    const responseData = await response.json();
+    if (!response.ok) throw new Error(responseData.status);
+  } catch (err) {
+    return err.message;
+  }
```

Comment:
This try-catch block still looks a little strange - the idea is right (passing the response code as error message for the component to display an appropriate message to the user), but the current code probably won't work. Try it - just call this function like `sendRequest('https://google.com', 'GET', {}, {})` at the end of the file, then run it directly with Node (`node src/utils/requests.js`).

There are two issues I see with this:
1. `responseData` is defined inside a try block, so it won't be accessible at the end of the function outside the try-catch. Refactor the code a bit.
2. In the case of response failure, if you're trying to throw an error with the response status (which is to be handled by the React component calling the function), then the above code won't work. The `throw new Error(...)` will just take the program flow to the catch block which will return the error message (which is not what we want).

Fix this, the rest seems good to go.